### DISCUSSION TITLE
Backup doesn't seem to work as documented

### DISCUSSION THREAD

**Discussion by anderwm** [Tecnativa/doodba#380] [Q&A] [answered]
This process
To restore backup:

```sh
# stop odoo if it's running
docker-compose stop odoo

# start backup and db
docker-compose up -d backup

# switch to some version
docker-compose exec backup restore --time TIME_IN_BACKUP_NAME --force

# ⚠️ DELETE PRODUCTION database
#docker-compose backup dropdb

# create new empty database
docker-compose exec backup createdb

# restore database
docker-compose exec backup sh -c 'psql -f $SRC/$PGDATABASE.sql'

# start odoo
docker-compose up -d
```
All goes well until reloading the .sql file. There are lots of errors on the db container like "thing already exists" or "violates foreign constraint" and "invalid command" and such. Then in the end I get a ir.http KeyError on the odoo container (with the 500 Internal Server Error).

Should I be running -i base somewhere in here? (like after createdb but before psql -f ...) (although I didn't immediately have success with this either). Or in general is there something I am missing?

**Comment by yajo** [comment, marked as answer]
That's because you didn't uncomment the line with `docker-compose backup dropdb`, which is commented on purpose because it's destructive.

**Reply by anderwm** [reply]
I actually did run dropdb, then made sure it was gone. These were still my results, I'll run through it a couple more times to make sure I didn't miss something.

**Reply by anderwm** [reply]
OK, wasn't using default "prod" database. Missed that I should do
# restore database
```docker-compose` exec backup sh -c 'psql -f $SRC/$PGDATABASE.sql -d desiredDBname'```
So it was trying to slam it all into "prod" DB.

Thanks for your time
